### PR TITLE
feat: Add imports to ServiceDeployment CRD

### PR DESCRIPTION
**Pull request by maciaszczykm**

### CODE REVIEW

File: controller/api/v1alpha1/servicedeployment_types.go [hunk 1]
```diff
@@ -109,12 +158,35 @@ type ServiceSpec struct {
 	// Templated should apply liquid templating to raw yaml files, defaults to true
 	// +kubebuilder:validation:Optional
 	Templated *bool `json:"templated,omitempty"`
-
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Imports are immutable"
+	Imports []corev1.ObjectReference `json:"imports"`
```
Comment (michaeljguarino) [inline controller/api/v1alpha1/servicedeployment_types.go:163]: I think to make this more future proof, we should have a type here like:

```
- stackRef:
      name: first-stack    
- stackRef:
      name: other-stack
```

That way we can add adjacent fields w/o breaking the schema

Comment (maciaszczykm) [reply controller/api/v1alpha1/servicedeployment_types.go]: Done.

**Review by michaeljguarino — APPROVED** [review]
Are we adding owner references to these resources?  Without requeueing we might not be able to gather changes on them easily

**Comment by maciaszczykm** [comment]
I'll add it.

**Comment by maciaszczykm** [comment]
Added and verified that it is set.

**Review by zreigz — APPROVED** [review]

**Pull request merged**